src/guest_memory.rs

```diff
@@ -513,6 +513,14 @@ pub trait GuestMemory {
         self.find_region(addr).map(|_| addr)
     }
 
+    /// Returns the GuestAddress(base + len) if the range [base, base + len) is valid.
+    fn check_range(&self, base: GuestAddress, len: usize) -> bool {
```

Comment:
I think a bool return value should be OK, because the typical usages are to validate the address range of a data structure. For example, when accessing a virtio descriptor, we could access individual fields after validating the whole descriptor is valid.